openedx/core/djangoapps/content_tagging/rest_api/v1/filters.py

```diff
@@ -2,20 +2,95 @@
 API Filters for content tagging org
 """
 
+from django.db.models import Exists, OuterRef, Q
 from rest_framework.filters import BaseFilterBackend
 
 import openedx_tagging.core.tagging.rules as oel_tagging
+from organizations.models import Organization
+
+from ...rules import get_admin_orgs, get_content_creator_orgs, get_instructor_orgs
+from ...models import TaxonomyOrg
 
 
 class UserOrgFilterBackend(BaseFilterBackend):
     """
+    Filter taxonomies based on user's orgs roles
+
     Taxonomy admin can see all taxonomies
-    Everyone else can see only enabled taxonomies
+    Org staff can see all taxonomies from their orgs
+    Content creators and instructors can see enabled taxonomies avaliable to their orgs
+    """
+
+    def filter_queryset(self, request, queryset, _):
+        if oel_tagging.is_taxonomy_admin(request.user):
+            return queryset
 
+        orgs = list(Organization.objects.all())
+        user_admin_orgs = get_admin_orgs(request.user, orgs)
+        content_creator_orgs = get_content_creator_orgs(request.user, orgs)
+        instructor_orgs = get_instructor_orgs(request.user, orgs)
+        user_orgs = list(set(content_creator_orgs) | set(instructor_orgs))
+
+        if len(user_orgs) == 0 and len(user_admin_orgs) == 0:
+            return queryset.none()
+
+        return queryset.filter(
+            # Get enabled taxonomies available to all orgs, or from orgs that the user is
+            # a content creator or instructor
+            Q(
+                Exists(
+                    TaxonomyOrg.objects
+                    .filter(
+                        taxonomy=OuterRef("pk"),
+                        rel_type=TaxonomyOrg.RelType.OWNER,
+                    )
+                    .filter(
+                        Q(org=None) |
+                        Q(org__short_name__in=[org.short_name for org in user_orgs])
```

Comment:
Since `org` is an actual Organization instance, this can just be:
```suggestion
                        Q(org__in=user_orgs)
```

Reply:
Done: db0229fe53d3d8b950cddb65f39edc69ce02cf95